internal/storage/observer.go

```diff
@@ -0,0 +1,326 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/mitchellh/copystructure"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/releaseutil"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+)
+
+// ObserverDriverName contains the string representation of Observer.
+const ObserverDriverName = "observer"
+
+var (
+	// ErrReleaseNotObserved indicates the release has not been observed by
+	// the Observator.
+	ErrReleaseNotObserved = errors.New("release: not observed")
+)
+
+// Observator reports about the write actions to a driver.Driver, recorded as
+// ObservedRelease objects.
+type Observator interface {
```

Comment:
Maybe it'd be nice to add a line about the origin of the name.
I think it's based on the interfaces that the `driver.Driver` embeds Creator, Updator, Deletor, Queryor.